### PR TITLE
DC/MLX5/FC: fixed resource leak

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixed resource leak in error handling: object send_op was
  leaked when fc grant operation was re-scheduled

### CODE REVIEW

**Review by dmitrygx — APPROVED** [review]

**Review by brminich — APPROVED** [review]

File: src/uct/ib/dc/dc_mlx5_ep.c [hunk 1]
```diff
@@ -993,8 +993,7 @@ uct_dc_mlx5_ep_fc_pure_grant_send_completion(uct_rc_iface_send_op_t *send_op,
     if (ucs_likely(!(send_op->flags & UCT_RC_IFACE_SEND_OP_STATUS) ||
                    (send_op->status != UCS_ERR_CANCELED))) {
         ucs_mpool_put(fc_req);
-        ucs_mpool_put(send_op);
-        return;
+        goto out;
```
[the remark this answers is not in the document]
Comment (hoopoepg) [reply src/uct/ib/dc/dc_mlx5_ep.c]: updated
I thought we don't like ```else```

**Review by dmitrygx — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Comment by yosefe** [comment]
@hoopoepg can you pls squash?

**Review by dmitrygx — APPROVED** [review]

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
MellanoxLab passed http://hpc-master.lab.mtl.com:8080/blue/organizations/jenkins/ucx/detail/ucx/14028/pipeline

**Pull request merged**